source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,351 @@ Documents
+     - When you need a type-safe API.
+
+   * - ``BasicDBObject``
+     - ``com.mongodb``
+     - No
+     - When you are migrating your application from a legacy driver version.
+
+While you can use any of the document classes in your application, we
+recommend that you use the ``Document`` class since it can concisely
+represent dynamically structured documents of any complexity. It
+implements the ``Map<String, Object>`` interface which enables it to use
+loosely-typed values.
+
+.. warning:: Duplicate Field Names
+
+   Although a document is valid if it contains two duplicate keys, the driver
+   only sets the value of the key to the value in the last instance.
 
 Document
 --------
 
-- Description
-- Chart that shows mapping between BSON and Java types (similar to the one here https://mongodb.github.io/mongo-java-driver/4.1/bson/documents/)
-- Code example of declaring one and appending fields
-- Code example of get() for various types
-- Link to Fundamentals > CRUD for more information
-- Link to API documentation
+The ``Document`` class offers a flexible representation of a BSON document.
+You can access and manipulate document data using Java types with this
+class. See the following table for mappings between frequently-used BSON
+and Java types:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - BSON type
+     - Java type
+   * - Array
+     - ``java.util.List``
+   * - Binary
+     - ``org.bson.types.Binary``
+   * - Boolean
+     - ``java.lang.Boolean``
+   * - Date
+     - ``java.util.Date``
+   * - Document
+     - ``org.bson.Document``
+   * - Double
+     - ``java.lang.Double``
+   * - Int32
+     - ``java.lang.Integer``
+   * - Int64
+     - ``java.lang.Long``
+   * - Null
+     - ``null``
+   * - ObjectId``
+     - ``org.bson.types.ObjectId``
+   * - String
+     - ``java.lang.String``
+
+You can also customize the type mapping by specifying a custom codec.
+For more information on customizing mapped types, see our guide on using
+:doc:`Codecs </fundamentals/data-formats/codecs>`.
+
+In the following code snippet, we show how to instantiate and build a sample
+``Document`` instance that contains a variety of data types:
+
+.. code-block:: java
+   :copyable: true
+
+   Document author = new Document("_id", new ObjectId())
+       .append("name", "Gabriel García Márquez")
+       .append("lastUpdated", new Date(120, 2, 26))
+       .append("novels", Arrays.asList(
+           new Document("title", "One Hundred Years of Solitude").append("yearPublished", 1967),
+           new Document("title", "Chronicle of a Death Foretold").append("yearPublished", 1981),
+           new Document("title", "Love in the Time of Cholera").append("yearPublished", 1985)));
+
+You can insert this document into a collection using the following code:
+
+.. code-block:: java
+   :copyable: true
+
+  // MongoClient mongoClient = ...
+
+  MongoDatabase database = mongoClient.getDatabase("test_data");
```

Comment:
Because Document is the default representation for database documents, it seems like you're just relying on the defaults here. Should we call this out explicitly above?

Reply:
The equivalent code snippets for other document representations use the "test_fundamentals" database instead of "test_data". Should we standardize the database used?

Reply:
> Because Document is the default representation for database documents, it seems like you're just relying on the defaults here. Should we call this out explicitly above?

I'm guessing you meant this on the next line for `getCollection`. I'll add a mention and link to the API docs.